ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -193,6 +195,17 @@ private KsqlEntity executeStatement(
             || statement instanceof DropStream
             || statement instanceof DropTable
     ) {
+      if (statement instanceof AbstractStreamCreateStatement) {
```

Comment:
nit: it probably makes sense to move this to a separate method called `maybeAddFieldsFromSchemaRegistry` . Would make the intent of this code block much clearer. As it stands, it is not clear that the statement is actually being changed here.